cluster.go

```diff
@@ -289,8 +292,11 @@ func (c *clusterClient) single() (conn conn) {
 func (c *clusterClient) nodes() []string {
 	c.mu.RLock()
 	nodes := make([]string, 0, len(c.conns))
-	for addr := range c.conns {
-		nodes = append(nodes, addr)
+	for addr, role := range c.conns {
+		// only display non-hidden connrole
+		if !role.hidden {
```

Comment:
We don't need to change this `nodes()` method. It is a private method for other testing purposes.

We should test this new `hidden` field explicitly in a new test case for the public `Node()` method.